src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -452,6 +455,20 @@ Error attachToSourceDatabase()
    FilePath existingSdb = sessionDirPath();
    if (existingSdb.exists())
       return sessionDirLock().acquire(sessionLockFilePath(existingSdb));
+   
+   // migrate from 'sdb' to current folder layout if needed
+   bool needsSdbMigration =
+         !sourceDatabaseRoot().exists() &&
+         sdbSourceDatabaseRoot().exists();
+   
+   if (needsSdbMigration)
```

Comment:
Since we don't blast the old source database, are we going to migrate from it every time we boot? Is that desired?